include/zephyr/net/socketcan.h

```diff
@@ -40,10 +40,24 @@ enum {
 };
 
 /* SocketCAN max data payload length */
+#ifndef CONFIG_CAN_FD_MODE
 #define SOCKETCAN_MAX_DLEN 8U
+#else
+#define SOCKETCAN_MAX_DLEN 64U
+#endif /* CONFIG_CAN_FD_MODE */
+
+/* SocketCAN MTU size compatible with Linux */
+#ifndef CONFIG_CAN_FD_MODE
+#define CAN_MTU (sizeof(struct socketcan_frame))
+#else
+#define CANFD_MTU (sizeof(struct socketcan_frame))
+#define CAN_MTU (CANFD_MTU - 56U)
+#endif
```

Comment:
I would prefer not negating the logic here - perhaps also collapse the two `#ifdef` sections?

```suggestion
/* SocketCAN MTU size compatible with Linux */
#ifdef CONFIG_CAN_FD_MODE
#define SOCKETCAN_MAX_DLEN 64U
#define CANFD_MTU (sizeof(struct socketcan_frame))
#define CAN_MTU (CANFD_MTU - 56U)
#else /* CONFIG_CAN_FD_MODE */
#define SOCKETCAN_MAX_DLEN 8U
#define CAN_MTU (sizeof(struct socketcan_frame))
#endif /* !CONFIG_CAN_FD_MODE */

Reply:
Minor suggestion, but I think it improves the code readability.

Reply:
Thanks. Done.